app/controllers/prescripteur_rdv_wizard_controller.rb

```diff
@@ -66,8 +67,30 @@ def create_rdv
     end
   end
 
+  def cancel_rdv
+    if @prescripteur.rdv.cancellable_by_user?
+      @prescripteur.rdv.update_and_notify(@prescripteur, status: "excused")
+      flash[:success] = "Le rendez-vous a bien été annulé."
+    else
+      flash[:error] = "Le rendez-vous ne peut plus être annulé."
+    end
+
+    redirect_to prescripteur_show_path
+  end
+
   def confirmation
-    @prescripteur = Prescripteur.find(session[:prescripteur_id])
+    render locals: { prescripteur: @prescripteur }
+  end
+
+  def show
+    if params[:token].present?
+      prescripteur = Prescripteur.find_by(token: params[:token])
```

Comment:
```suggestion
      prescripteur = Prescripteur.find_by!(token: params[:token])
```

Je pense qu’il vaut mieux que ça échoue explicitement ici sinon ça va faire une nil error à la ligne d’en dessous

Reply:
Bien vu !